### PR TITLE
[REEF-451] Add Org.Apache.REEF.IO

### PR DESCRIPTION
**Pull request by markusweimer**
This creates the new (empty) project Org.Apache.REEF.IO to hold our data and filesystem abstractions.

JIRA:  [REEF-451](https://issues.apache.org/jira/browse/REEF-451)

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    <owners>The Apache REEF project</owners>
+    <licenseUrl>http://www.apache.org/licenses/LICENSE-2.0</licenseUrl>
+    <projectUrl>http://reef.incubator.apache.org/</projectUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>I/O Library for Apache REEF</description>
+    <copyright>The Apache Software Foundation</copyright>
+    <tags>REEF IO</tags>
+  </metadata>
+  <files>
+      <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.IMRU\Org.Apache.REEF.IO.dll" target="lib\net45" />
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec]: Should be Org.Apache.REEF.IO

File: lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      <file src="..\bin\$Platform$\$Configuration$\Org.Apache.REEF.IMRU\Org.Apache.REEF.IO.pdb" target="lib\net45" />
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec]: same as above

File: lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.csproj [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <ProjectGuid>{DEC0F0A8-DBEF-4EBF-B69C-E2369C15ABF1}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Org.Apache.REEF.IO</RootNamespace>
+    <AssemblyName>Org.Apache.REEF.IO</AssemblyName>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..</SolutionDir>
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.csproj]: add <RestorePackages>true</RestorePackages>

Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.csproj]: done.

File: lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec (hunk 1 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec]: Done.

File: lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec (hunk 2 above)
Comment (markusweimer) [inline lang/cs/Org.Apache.REEF.IO/Org.Apache.REEF.IO.nuspec]: Done.

**Comment by markusweimer** [comment]
@jwang98052 thanks for the review, I addressed your comments.

**Pull request closed without merging**